### ISSUE TITLE
Add support for logging and visualizing the loss and learning rate with tensorboard.

### ISSUE THREAD

**Issue by ZeroCool940711** [lucidrains/muse-maskgit-pytorch#23]
Hi there, it would be nice to have support for tensorboard, have it so the loss and learning rate values are logged so we can use tensorboard to keep track of them and get a better idea of how the model changes overtime. It would also be nice for having an idea of how many steps we have trained for as it's really easy to lose track of the global number of steps you have trained for unless you do everything on a single session. I think it would be nice if we could have a global step counter that would persist through sessions, this could also be done with tensorboard or even easier by just having a json file with the info of the training session so we can reuse it for the next session, this would also make it easier to resume a session, we can just load the vae and continue from it but for the trainer it is as if we would be training from scratch every time.

**Comment by scarbain** [comment]
Hi @ZeroCool940711 
Here's the PR : https://github.com/Sygil-Dev/muse-maskgit-pytorch/pull/7

**Comment by ZeroCool940711** [comment]
@scarbain Thanks for the contribution to our fork, I think once we have the training script working properly there we can make a PR here to lucidrains' repo to keep things in sync and contribute to this repo too :)
